vllm/spec_decode/multi_step_worker.py

```diff
@@ -11,6 +12,68 @@
 from vllm.worker.worker import Worker
 
 
+class NGramDrafter():
```

Comment:
Can we break ngram stuff out into its own file? basically, the file `multi_step_worker.py` can be specialized to draft model, then we can have `ngram_worker.py` in its own file.

Reply:
Add ngram_worker.py in the latest commit, also I refactor DraftModelTop1Proposer out, and ngram_worker/multi_step_worker current share the same logic.